[RFR] add README

ThieryMichel opened the pull request.

ThieryMichel committed:
add README

ThieryMichel committed:
more generic createEffect

djhi commented on README.md:
```diff
@@ -1,2 +1,105 @@
-SG
-==
+#SG
+Utility to handle side effects with generator. Inspired by redux-saga and co.
```
links to saga and co ?

ThieryMichel committed:
complete README

ThieryMichel committed:
add link to redux-saga and co

djhi commented on README.md:
```diff
@@ -23,23 +23,19 @@ func('world')
-An effect is an object which describe what to do.
-This allow to test the generator flow logic without having to worry about the effect.
+##effects
+The effects are helper who return object describing what to do.
```
s/helper/helpers/

djhi commented on README.md:
```diff
@@ -1,2 +1,101 @@
-SG
-==
+#SG
+Utility to handle side effects with generator. Inspired by [redux-saga](https://github.com/yelouafi/redux-saga) and [co](https://github.com/tj/co).
```
s/generator/generators

djhi replied:
s/generator/generators

djhi replied:
s/generator/generators/

djhi commented on README.md:
````diff
@@ -1,2 +1,101 @@
+```
+
+##effects
+The effects are helper who return object describing what to do.
````
s/who/which/

djhi commented on README.md:
```diff
@@ -1,2 +1,101 @@
+
+###Adding your own custom effects with createEffect
+You can create your own effect with createEffect
+It take a type and an handler and an optional context.
```
s/a type and an handler/a type, an handler/

djhi commented on README.md:
```diff
@@ -1,2 +1,101 @@
+You can create your own effect with createEffect
+It take a type and an handler and an optional context.
+Type: string with the effect name
+Handler: a function that will be called with the effect parameter and must return a promise or be async.
```
s/parameter/parameters/

djhi commented on README.md:
```diff
@@ -1,2 +1,101 @@
+It take a type and an handler and an optional context.
+Type: string with the effect name
+Handler: a function that will be called with the effect parameter and must return a promise or be async.
+Context: the context(this value) with which the handler will be called. (context will have no effect with arrow function)
```
s/context(this/context (this/

sedy-bot committed:
Typo fix s/who/which/

As requested by @djhi at https://github.com/marmelab/sg/pull/6#discussion_r90187595

sedy-bot committed:
Typo fix s/a type and an handler/a type, an handler/

As requested by @djhi at https://github.com/marmelab/sg/pull/6#discussion_r90187960

djhi committed:
Update README.md

Fix typos

djhi committed:
Update README.md

fix typo

djhi committed:
Update README.md

better sentences

djhi committed:
Update README.md

better english
